### PR TITLE
Cleanup ocp4_workload-authentication role formatting

### PR DESCRIPTION
**Pull request by ritesh97-rh**
##### SUMMARY
For the specified role:
- added yamllint file to ocp4_workload_authentication role
- cleaned code up as per formatting requirements
- tightened formatting warnings into errors for the future

##### ISSUE TYPE
- Code Cleanup Request

##### COMPONENT NAME
ocp4_workload_authentication

### CODE REVIEW

File: ansible/roles_ocp_workloads/ocp4_workload_authentication/tasks/workload.yml [hunk 1]
```diff
@@ -3,7 +3,10 @@
   assert:
     that:
     - ocp4_workload_authentication_idm_type is defined
-    - ocp4_workload_authentication_idm_type == "none" or ocp4_workload_authentication_idm_type == "htpasswd" or ocp4_workload_authentication_idm_type == "ldap"
+    - >-
+      ocp4_workload_authentication_idm_type == "none"
+      or ocp4_workload_authentication_idm_type == "htpasswd" or
+      ocp4_workload_authentication_idm_type == "ldap"
```
Comment (jkupferer) [inline ansible/roles_ocp_workloads/ocp4_workload_authentication/tasks/workload.yml:9]: Reformat to be consistent with `or` placement.

Ex:
```
    - >-
      ocp4_workload_authentication_idm_type == "none" or
      ocp4_workload_authentication_idm_type == "htpasswd" or
      ocp4_workload_authentication_idm_type == "ldap"
```

File: ansible/roles_ocp_workloads/ocp4_workload_authentication/tasks/workload.yml [hunk 2]
```diff
@@ -12,7 +15,10 @@
   - name: Generate cluster admin password
     when: ocp4_workload_authentication_htpasswd_admin_password | default('') | length == 0
     set_fact:
-      ocp4_workload_authentication_admin_password: "{{ lookup('password', '/dev/null length={{ ocp4_workload_authentication_htpasswd_admin_password_length }} chars=ascii_letters,digits') }}"
+      ocp4_workload_authentication_admin_password: >-
+        {{ lookup('password',
+        '/dev/null length={{ ocp4_workload_authentication_htpasswd_admin_password_length }}
+        chars=ascii_letters,digits') }}
```
Comment (jkupferer) [inline ansible/roles_ocp_workloads/ocp4_workload_authentication/tasks/workload.yml:21]: Hrm, this looks really weird... The way lookup plugins take arguments is weird... but how about:

```
      ocp4_workload_authentication_admin_password: >-
￼        {{ lookup('password', '/dev/null chars=ascii_letters,digits '
￼            ~ 'length=' ~ ocp4_workload_authentication_htpasswd_admin_password_length
￼        ) }}
```

File: ansible/roles_ocp_workloads/ocp4_workload_authentication/tasks/workload.yml [hunk 3]
```diff
@@ -33,7 +39,10 @@
   - name: Generate user passwords
     when: ocp4_workload_authentication_htpasswd_user_password | default('') | length == 0
     set_fact:
-      ocp4_workload_authentication_user_password: "{{ lookup('password', '/dev/null length={{ ocp4_workload_authentication_htpasswd_user_password_length }} chars=ascii_letters,digits') }}"
+      ocp4_workload_authentication_user_password: >-
+        {{ lookup('password',
+        '/dev/null length={{ ocp4_workload_authentication_htpasswd_user_password_length }}
+        chars=ascii_letters,digits') }}
```
Comment (jkupferer) [inline ansible/roles_ocp_workloads/ocp4_workload_authentication/tasks/workload.yml:45]: Same here... seeing `{{` inside of `{{` really looks weird and wrong.

File: ansible/roles_ocp_workloads/ocp4_workload_authentication/tasks/workload.yml [hunk 4]
```diff
@@ -89,15 +100,22 @@
     loop:
     - ""
     - "HTPasswd Authentication is enabled on this cluster."
-    - "Users {{ ocp4_workload_authentication_htpasswd_user_base }}1 .. {{ ocp4_workload_authentication_htpasswd_user_base }}{{ ocp4_workload_authentication_htpasswd_user_count }} are created with password `{{ ocp4_workload_authentication_user_password }}`"
-    - "User `{{ ocp4_workload_authentication_admin_user }}` with password `{{ ocp4_workload_authentication_admin_password }}` is cluster admin."    
+    - >-
+      Users {{ ocp4_workload_authentication_htpasswd_user_base }}1 ..
+      {{ ocp4_workload_authentication_htpasswd_user_base }}{{ ocp4_workload_authentication_htpasswd_user_count }} are
```
Comment (jkupferer) [inline ansible/roles_ocp_workloads/ocp4_workload_authentication/tasks/workload.yml:105]: This is weird in the original, but would be better with string concatenation operator, `~`: 

```
{{ ocp4_workload_authentication_htpasswd_user_base ~ ocp4_workload_authentication_htpasswd_user_count }}
```

File: ansible/roles_ocp_workloads/ocp4_workload_authentication/tasks/workload.yml [hunk 5]
```diff
@@ -136,7 +159,10 @@
       namespace: openshift-config
 
   - name: Create LDAP Bind Password Secret
-    shell: "oc create secret generic opentlc-ldap-secret --from-literal=bindPassword=\"{{ ocp4_workload_authentication_ldap_bind_password }}\" -n openshift-config"
+    shell: >-
+      oc create secret generic opentlc-ldap-secret
+      --from-literal=bindPassword=\"{{ ocp4_workload_authentication_ldap_bind_password }}\"
```
Comment (jkupferer) [inline ansible/roles_ocp_workloads/ocp4_workload_authentication/tasks/workload.yml:164]: No `\` needed here with the changed quote style.

**Review by jkupferer — CHANGES_REQUESTED** [review]

File: ansible/roles_ocp_workloads/ocp4_workload_authentication/tasks/workload.yml (hunk 2 above)
Comment (ritesh97-rh) [reply ansible/roles_ocp_workloads/ocp4_workload_authentication/tasks/workload.yml:21]: It doesn't seem to work.
syntax error: could not find expected ':' (syntax) is the error I'm receiving

Comment (ritesh97-rh) [reply ansible/roles_ocp_workloads/ocp4_workload_authentication/tasks/workload.yml]: Never mind, figured it out.

**Review by jkupferer — APPROVED** [review]

**Pull request merged**